packages/x-date-pickers/src/internals/hooks/usePicker/usePicker.tsx

```diff
@@ -223,13 +241,15 @@ export const usePicker = <
     return 'enabled';
   }, [disableOpenPicker, hasUIView, disabled, readOnly]);
 
-  const stepNavigation = getStepNavigation({
-    setView,
-    view,
-    initialView: initialView ?? views[0],
-    views,
-  });
-  const wrappedGoToNextStep = useEventCallback(stepNavigation.goToNextStep);
+  const wrappedGoToNextStep = useEventCallback(goToNextStep);
+
+  const defaultActionBarActions = React.useMemo<PickersActionBarAction[]>(() => {
+    if (closeOnSelect && !hasSeveralSteps) {
```

Comment:
This causes a behavioral change.
I.e. If `closeOnSelect={false}` on `DesktopDateRangePicker` - we get the `cancel` and `ok` actions. 🤔 

Are we ok with this? 🤔 
I feel like the good old duplication where necessary is sometimes safer. 🙈 😆

Reply:
I feel like the new behavior makes more sense
Otherwise, `closeOnSelect={false}` on a `DesktopDateRangePicker` is basically broken without manually overriding the actions.